scripts/convert_hf_checkpoint.py

```diff
@@ -83,66 +78,57 @@ def permute(w):
-                if "rotary_emb.inv_freq" in name:
-                    continue
-                if "model.layers" in name:
-                    block_id = int(name.split(".")[2])
-                    from_name = ".".join(name.split(".")[3:])
-                    to_name = weight_map[from_name]
-                    sd_key = f"transformer.h.{block_id}.{to_name}"
-
-                    if "q_proj" in name:
-                        unprocessed_weights[sd_key]["q_proj"] = param
-                        skip = True
-                    elif "k_proj" in name:
-                        unprocessed_weights[sd_key]["k_proj"] = param
-                        skip = True
-                    elif "v_proj" in name:
-                        unprocessed_weights[sd_key]["v_proj"] = param
-                        skip = True
+            with lazy_load(bin_file) as hf_weights:
+                for name, param in hf_weights.items():
+                    skip = False
+                    if "rotary_emb.inv_freq" in name:
+                        continue
+                    if "model.layers" in name:
+                        block_id = int(name.split(".")[2])
+                        from_name = ".".join(name.split(".")[3:])
+                        to_name = weight_map[from_name]
+                        sd_key = f"transformer.h.{block_id}.{to_name}"
+
+                        if "q_proj" in name:
+                            unprocessed_weights[sd_key]["q_proj"] = param
+                            skip = True
+                        elif "k_proj" in name:
+                            unprocessed_weights[sd_key]["k_proj"] = param
+                            skip = True
+                        elif "v_proj" in name:
+                            unprocessed_weights[sd_key]["v_proj"] = param
+                            skip = True
+                        if skip and len(unprocessed_weights[sd_key]) == 3:
+                            w = torch.empty(
+                                sd_meta[sd_key].shape, dtype=sd_meta[sd_key].dtype
+                            )
+                            w[:qkv_size] = permute(unprocessed_weights[sd_key]["q_proj"])
+                            w[qkv_size:-qkv_size] = permute(
+                                unprocessed_weights[sd_key]["k_proj"]
+                            )
+                            w[-qkv_size:] = (
+                                unprocessed_weights[sd_key]["v_proj"]
+                                ._load_tensor()
+                                .to(dtype)
+                            )
+                            sd[sd_key] = w
+                            del unprocessed_weights[sd_key]
+                            skip = False
+                        else:
+                            sd[sd_key] = param._load_tensor().to(dtype)
                     else:
+                        sd_key = weight_map[name]
                         sd[sd_key] = param._load_tensor().to(dtype)
```

Comment:
This else in the previous code is now removed, it should fall under the L121 else in the new code